src/lib/renderer/decorator/autocomplete/AutocompleteBehaviour.ts

```diff
@@ -0,0 +1,79 @@
+import {Defaults} from "../../../support/Defaults";
+import {IIndex, IBehaviour} from "../../../api";
+import {FlexSearchIndex} from "./index/FlexSearchIndex";
+import {AutocompleteDropdown} from "./AutocompleteDropdown";
+import {Subscription} from "../../../connection/Subscription";
+
+/**
+ * Configuration for enabling Autocompletion for the text input field.
+ *
+ * @member {IAutocompleteElementProvider} elementProvider is used to load all possible autocomplete values initially.
+ * @member {HTMLTextAreaElement} textInput is used as basis for the search for fitting auto completions.
+ * Per default the first element with class lto-textbox is used.
+ * @member {HTMLDivElement} dropdownElement is used as container to render the autocompletion dropdown.
+ * Per default the first element with class lto-autocomplete is used.
+ * @member {number} maxNumberOfResults is used to determine the max elements displayed in the autocomplete dropdown.
+ * Default is 5.
+ * @member {IIndex} index is used to store and search for autocompletion elements.
+ */
+interface AutocompleteConfig {
+    elements: (() => Promise<string[]>)[];
+    dropdownElement?: HTMLDivElement;
+    invokerElement?: HTMLButtonElement;
+    maxNumberOfResults?: number;
+    index?: IIndex;
+}
+
+/**
+ * The autocomplete decorator adds an autocomplete mechanism
+ * for the user input.
+ */
+export class AutocompleteBehaviour extends IBehaviour {
+    private readonly index: IIndex;
+    private readonly autocompleteDropdown: AutocompleteDropdown;
+    private readonly maxNumberOfResults: number;
+
+    constructor(config: AutocompleteConfig, textbox: HTMLTextAreaElement = Defaults.textbox()) {
+        super(textbox, [
+            {type: "input", handler: () => this.onKeydown(textbox)}
+        ]);
+        this.index = config.index || new FlexSearchIndex();
+        this.maxNumberOfResults = config.maxNumberOfResults || 5;
+        this.autocompleteDropdown = new AutocompleteDropdown(config.dropdownElement || Defaults.autocomplete(), (event: Event) => {
+            textbox.value = (event.currentTarget as HTMLElement).innerText;
+            (config.invokerElement || Defaults.invoker()).click();
+        });
+        this.addElements(config.elements);
+    }
+
+    private addElements(elementFactories: (() => Promise<string[]>)[]) {
```

Comment:
Ich glaub ich bin blind :-) wo werden die Intents bzw. elements ausfiltriert? d.h. elementen die "complete='false'" sollten nicht in dem AutocompleteDropdown hinzufügen. Danke

Reply:
Das Autocomplete ist generisch gehalten - Intents müssen vom Benutzer eingepflegt werden.